lib/vanagon/project.rb

```diff
@@ -476,5 +477,36 @@ def generate_package
     def generate_packaging_artifacts(workdir)
       @platform.generate_packaging_artifacts(workdir, @name, binding)
     end
+
+    # Generate a json file which lists all of the dependant components of the project
+    def generate_dependencies_info
+      project_components = @components.map do |component|
+        components_hash = component.options
+        [:version, :url].each do |key|
+          if val = component.send(key)
+            components_hash[key] = val
+          end
+        end
+        [
+          component.name,
+          components_hash,
+        ]
+      end.to_h
+    end
+
+    def generate_manifest_json
+      build_time = Time.now
+      manifest = {
+        "packaging_type" => {
+          "vanagon" => Gem.loaded_specs["vanagon"].version.to_s,
+        },
+        "git_sha" => @version,
+        "components" => generate_dependencies_info,
+        "build_time" => build_time.rfc2822,
+      }
+      File.open(File.join('ext', 'build_metadata.json'), 'w') do |f|
```

Comment:
The `./ext` directory should exist in the puppetlabs projects. Maybe I should create it here in case it doesn't exist in non-pl projects? Or conditionally put this file in `ext` if that directory does exist.

It is currently there because the packaging gem will pick up the file in this location and ship it to the build server.